### PR TITLE
Try using a faster class check

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
Adding a call that's not used outside of String bridging stuff definitely did not regress Array2D

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
Hah, thought so!

**Comment by Catfish-Man** [comment]
@swift-ci please test

File: stdlib/public/core/Builtin.swift [hunk 1]
```diff
@@ -323,12 +323,25 @@ internal func _branchHint(_ actual: Bool, expected: Bool) -> Bool {
...
   return true
 }
+
+@_effects(readonly)
+@_silgen_name("swift_stdlib_getObjCClassOfAllocated")
+internal func _stdlib_classOfObjCHeapObject(_ obj: AnyObject) -> AnyClass {
+  fatalError(
+    "Asking for the ObjC class of an object on non-ObjC runtimes is unsupported"
+  )
+}
```
Comment (milseman) [inline stdlib/public/core/Builtin.swift]: Is this necessary? If you omit it, then we get a compile-time rather than run-time error.

**Review by milseman — COMMENTED** [review]
Neat, any idea how much faster?

File: stdlib/public/core/Builtin.swift (hunk 1 above)
Comment (Catfish-Man) [reply stdlib/public/core/Builtin.swift]: Oh, right, good point

**Comment by Catfish-Man** [comment]
> Neat, any idea how much faster?

Only like 3% or so, but I like not leaving minor perf hazards laying around.

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
@swift-ci please test

File: stdlib/public/core/Builtin.swift [hunk 2]
```diff
@@ -323,6 +323,11 @@ internal func _branchHint(_ actual: Bool, expected: Bool) -> Bool {
 @usableFromInline
 @_silgen_name("_swift_objcClassUsesNativeSwiftReferenceCounting")
 internal func _usesNativeSwiftReferenceCounting(_ theClass: AnyClass) -> Bool
+
+/// Returns the class of a non-tagged-pointer Objective-C object
+@_effects(readonly)
+@_silgen_name("swift_stdlib_getObjCClassOfAllocated")
+internal func _stdlib_classOfObjCHeapObject(_ object: AnyObject) -> AnyClass
```
Comment (mikeash) [inline stdlib/public/core/Builtin.swift]: This naming inconsistency is bothersome. Otherwise seems good.

Comment (Catfish-Man) [reply stdlib/public/core/Builtin.swift]: Renamed everything to _swift_classOfObjCHeapObject. That work? :)

**Review by mikeash — APPROVED** [review]

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test and merge

**Pull request merged**